src/core/lib/security/authorization/rbac_translator.cc

```diff
@@ -251,61 +267,73 @@ absl::StatusOr<Rbac::Permission> ParseRequest(const Json& json) {
   return Rbac::Permission::MakeAndPermission(std::move(request));
 }
 
-absl::StatusOr<Rbac::Policy> ParseRules(const Json& json) {
-  Rbac::Principal principals;
-  auto it = json.object_value().find("source");
-  if (it != json.object_value().end()) {
-    if (it->second.type() != Json::Type::OBJECT) {
-      return absl::InvalidArgumentError("\"source\" is not an object.");
+absl::StatusOr<Rbac::Policy> ParseRule(const Json& json,
+                                       std::string* policy_name) {
+  std::unique_ptr<Rbac::Principal> principals;
```

Comment:
Do these need to be unique_ptrs, or can they just be local variables?  It doesn't look like we're actually depending on the fact that they are dynamically allocated.

Reply:
I am using unique_ptr here, so we can check later if "source" or "request" field is absent.

Reply:
If the goal is to check whether the fields are set, then how about using `absl::optional<>` instead?  That way you get the same semantics, but you don't need any dymamic allocation.